Feature: support NVTX profiling via timer_enable_nvtx flag

wangtianxiang opened the pull request:
### What's changed?
This PR adds **NVTX range markers** inside `timer::tick()`, enabling clear visualization of timer scopes in GPU profiling tools like **Nsight Systems**.
### 🎯 Why This Matters

- ✅ **Visual correlation** between timer labels and GPU activities — see exactly what runs on CPU vs GPU.
- 🔍 Easily identify **CPU-bound sections** (e.g. `Charge_Mixing::mixing_rho`, `Potential::cal_veff`) that are candidates for GPU porting.
- ⏱️ Timing ranges appear as colored blocks in timeline — showing **duration, overlap, and gaps** with CUDA kernels.

> 💡 Example: In the screenshot below, you can see `Charge_Mixing::mixing_rho` and `Potential::cal_veff` running purely on CPU — no GPU activity underneath — indicating clear optimization/porting opportunities.
<img width="1586" height="793" alt="image" src="https://github.com/user-attachments/assets/d4d80b1d-3601-4ee3-a9fc-b55fd592881e" />

---

### ⚙️ How to Use

Enable NVTX labeling by adding `timer_enable_nvtx` to the `INPUT` file:

```bash
timer_enable_nvtx  1
```

Signed-off-by：Tianxiang Wang<tianxiang.wang@metax-tech.com>, Contributed under MetaX Integrated Circuits (Shanghai) Co., Ltd.

wangtianxiang committed:
Feature: support NVTX profiling via timer_enable_nvtx flag
Signed-off-by：Tianxiang Wang<tianxiang.wang@metax-tech.com>, Contributed under MetaX Integrated Circuits (Shanghai) Co., Ltd.

wangtianxiang committed:
Add timer_enable_nvtx section in markdown
Signed-off-by：Tianxiang Wang<tianxiang.wang@metax-tech.com>, Contributed under MetaX Integrated Circuits (Shanghai) Co., Ltd.

mohanchen commented on docs/advanced/input_files/input-main.md:
```diff
@@ -706,6 +707,15 @@ If only one value is set (such as `kspacing 0.5`), then kspacing values of a/b/c
   - double: double precision
 - **Default**: double
 
+### timer_enable_nvtx
```
is this parameter only valid for CUDA environment?

mohanchen replied:
if so, we should clarify this in the document

wangtianxiang replied:
Yes, this parameter is only valid in CUDA environments.
We’ve already added a note in the documentation to clarify this.

wangtianxiang committed:
Fix: Use __USE_NVTX macro to avoid NVTX linking errors in tests.
Clarify in docs that timer_enable_nvtx parameter only takes effect on CUDA platforms.
Signed-off-by：Tianxiang Wang<tianxiang.wang@metax-tech.com>, Contributed under MetaX Integrated Circuits (Shanghai) Co., Ltd.

wangtianxiang committed:
Merge branch 'develop' into add_nvtx

mohanchen commented:
This is a nice feature, thanks for your contribution!
